webapp/channels/src/utils/constants.tsx

```diff
@@ -2009,7 +2009,7 @@ export const Constants = {
     SEARCH_POST: 'searchpost',
     CHANNEL_ID_LENGTH: 26,
     TRANSPARENT_PIXEL: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=',
-    TRIPLE_BACK_TICKS: /```/g,
+    REGEX_CODE_BLOCK_OPTIONAL_LANGUAGE_TAG: /^```.*$/gm,
```

Comment:
Could you remove the `g` from this? I think that was another bug in the existing regex since it means that the regex will look for look for more matches if you call it on the same text multiple times, and that means that it'll return different results if it happens to be called on the same string multiple times in a row
![Screenshot 2023-09-15 at 12 17 48 PM](https://github.com/mattermost/mattermost/assets/3277310/cd8792ac-2ff4-4ee4-a726-8c4d13c66e7a)

Reply:
> Could you remove the `g` from this? I think that was another bug in the existing regex since it means that the regex will look for look for more matches if you call it on the same text multiple times, and that means that it'll return different results if it happens to be called on the same string multiple times in a row ![Screenshot 2023-09-15 at 12 17 48 PM](https://user-images.githubusercontent.com/3277310/268338770-cd8792ac-2ff4-4ee4-a726-8c4d13c66e7a.png)

Thanks for the feedback! You're spot-on about the g flag affecting RegExp.prototype.test(). However, we're using String.prototype.match(), which doesn't show that behavior. I've gone ahead and added a test to confirm the consistency of the results. For further reference, here's the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match#description).

Reply:
Ah, so it is. I guess I shouldn't be surprised that two very similar APIs behave differently since this is JavaScript after all 😅

Good idea to add a test to confirm that